plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -665,26 +665,21 @@ static void vaapi_defaults_internal(obs_data_t *settings, enum codec_type codec)
 	const char *const device = vaapi_default_device(codec);
 	obs_data_set_default_string(settings, "vaapi_device", device);
 #ifdef ENABLE_HEVC
-	if (codec == CODEC_HEVC) {
+	if (codec == CODEC_HEVC)
 		obs_data_set_default_int(settings, "profile",
 					 FF_PROFILE_HEVC_MAIN);
-		obs_data_set_default_int(settings, "level", 120);
-
-	} else
+	else
 #endif
-		if (codec == CODEC_H264) {
+		if (codec == CODEC_H264)
 		obs_data_set_default_int(settings, "profile",
-					 FF_PROFILE_H264_CONSTRAINED_BASELINE);
```

Comment:
This is historically here to avoid corrupted h264 bitstreams from broken amd cards... though ive still seen one such card in support even recently maybe we can finally remove this without affecting too many people.

Reply:
If this is broken on some cards, it should be fixed in drivers instead.

Reply:
You would think but amd wasnt interested so here we are.

Reply:
Is there a mesa issue for this?

Reply:
https://github.com/obsproject/obs-studio/pull/1256#issuecomment-380642787, is the easiest to find, but unless mesa's old bug tracker is archived i doubt you would have a good time finding the issue.

Reply:
All issues were migrated from old bug tracker to gitlab, but I still can't find any related issue. Also I still don't see what generation of cards would be affected from that comment. In any case, the encoding support in mesa has been greatly improved since 2018.

Reply:
Tested on RX 570 now and it works fine.

Reply:
Thanks we can see if people report issues later during beta/rc, I guess.